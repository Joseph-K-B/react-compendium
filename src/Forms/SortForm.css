.sortSection {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.sortOrder {
    display: flex;
    flex-direction: column;
    margin: 5%;
}

.sortStat {
    position: relative;
    display: flex;
    flex-direction: column;
    text-align: center;
    margin: 5%;
}

.sortStat select {
    color: white;
    background-color: #393E41;
    box-shadow: 0 0 5px 3px red;
    text-align: center;
    height: 2rem;
    width: 7rem;
    -webkit-appearance: none;
}

.sortStat select:hover {
    color: black;
    background-color: #D3D0CB;
}