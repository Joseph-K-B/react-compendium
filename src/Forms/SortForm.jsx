import './SortForm.css';


function SortForm ({
    sortOrder, 
    sortStat,
    sortChange,
    statChange,
    handleReset
}) {
    return (
        <>
        <section className='sortSection'>
            <div aria-label='sortOrder' className='sortOrder'>
                <h3>Order</h3>               
                <label value='asc'>Ascending</label>
                <input
                    type='radio' 
                    value='asc' 
                    onChange={(e) => sortChange(e.target.value)}
                    checked={sortOrder === 'asc'}
                >
                </input>
                <label value='desc'>Descending</label>
                <input
                    type='radio' 
                    value='desc' 
                    onChange={(e) => sortChange(e.target.value)}
                    checked={sortOrder === 'desc'}    
                >
                </input>
            </div>
            <div aria-label='sortStat' className='sortStat'>
                <h3>Stats</h3>
                <select
                    className='dropDown' 
                    value={sortStat} 
                    onChange={(e) => statChange(e.target.value)}>
                        <option value='attack'>Attack</option>
                        <option value='defense'>Defense</option>
                        <option value='hp'>HP</option>
                        <option value='speed'>Speed</option>
                </select>
            <button onClick={handleReset}>Reset</button>   
            </div>
        </section>
        </>
    )
}

export default SortForm