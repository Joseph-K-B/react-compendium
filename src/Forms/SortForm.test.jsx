import { render, screen } from "@testing-library/react";
import SortForm from "./SortForm";

it('should render sort form controls', () => {
    const {container} = render(<SortForm/>);
    const sortOrder = screen.getByLabelText(/sortOrder/);
    const sortStat = screen.getByLabelText(/sortStat/);

    expect(sortOrder).toBeInTheDocument();
    expect(sortStat).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});