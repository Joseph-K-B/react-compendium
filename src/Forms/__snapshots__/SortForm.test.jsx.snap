// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render sort form controls 1`] = `
<div>
  <section
    class="sortSection"
  >
    <div
      aria-label="sortOrder"
      class="sortOrder"
    >
      <h3>
        Order
      </h3>
      <label
        value="asc"
      >
        Ascending
      </label>
      <input
        type="radio"
        value="asc"
      />
      <label
        value="desc"
      >
        Descending
      </label>
      <input
        type="radio"
        value="desc"
      />
    </div>
    <div
      aria-label="sortStat"
      class="sortStat"
    >
      <h3>
        Stats
      </h3>
      <select
        class="dropDown"
      >
        <option
          value="attack"
        >
          Attack
        </option>
        <option
          value="defense"
        >
          Defense
        </option>
        <option
          value="hp"
        >
          HP
        </option>
        <option
          value="speed"
        >
          Speed
        </option>
      </select>
      <button>
        Reset
      </button>
    </div>
  </section>
</div>
`;
